data/codex_related.py

```diff
@@ -75,20 +79,57 @@ def process_datapoint(self,datapoint:dict):
             
             answer_body = accepted_answer["body"] #Get the body of the accepted answer
             accepted_answer_code : list[str] = parse_html_block_for_code(answer_body)
-            
             return accepted_answer_code
-
-    def apply_codex_for_type_inference(self,datapoint:str)->dict:
-        pass
-
-    def iter_apply_process(self,length:int):
+        else:
+            return []
+
+    def apply_codex_for_type_inference(self,input_prompt:str)->dict:
+        """
+        Apply codex for type annotation of the input prompt
+
+        args:
+            input_prompt (str) : input prompt to be annotated
+        """
+        response = openai.Edit.create(
+        model = "code-davinci-edit-001",
```

Comment:
@reshinthadithyan don't see this model [listed](https://beta.openai.com/docs/models) anymore. Is it the model you want to use? I presume you want to use `code-davinci-002`? Just curious to know

Reply:
It's weird that the model doesn't exist there. But check [here](https://github.com/openai/openai-cookbook#3-edit-code), it's mentioned. Might be because the model is in private beta.

Reply:
Yeah, I noticed that. I have a private beta account and already attempted to use that model. Always give me `Rate limit exceeded. Sounds like you are able to invoke API calls to that model. Just curious to know.